### DISCUSSION TITLE
SOLVED - How do I add gpg keys from Windows 10 and WSL

### DISCUSSION THREAD

**Discussion by jaytelford** [community/community#21604] [New to GitHub]
<p>Hi, </p>
<p>I have been trying to add a GPG key created on the Windows Subsystem for Linux (WSL) platform.  I have created keys as per <a href="https://help.github.com/articles/generating-a-new-gpg-key/" rel="nofollow noopener">these instructions</a> with both GPGv1 and GPGv2.  They keys are created perfectly fine but I cant add them to GitHub.  When I try to add a new GPG key <a href="https://help.github.com/articles/adding-a-new-gpg-key-to-your-github-account/" rel="nofollow noopener">as instructed in this help article</a> I keep getting an error informing me that there was an “error doing that” and when the page reloads after save, they key hasnt been added.</p>
<p>I have a working GPG key that I added from my Mac but thats obviously just for use on that machine.  However, I also use WSL, Windows 10 and Debian 8 server.  I need to be able to use my GPG signature on all systems.</p>
<p>Can anyone offer any help.  How do I add GPG keys from WSL and Windows 10?</p>
<p>Cheers</p>
<p>Jay</p>

**Comment by jaytelford** [comment]
<p>One thought crossed my mind.  Can I use the same email address on multiple GPG keys?  Come to think of it, maybe not.  I will try changing the email and doing it again.</p>

**Comment by jaytelford** [comment, marked as answer]
<p>This has not been solved.  It had to do with the way Win10 pastes stuff.  To fix this I had to use pastebin, paste into their then correct the formatting - copy as plain text and then paste as plain text into the field in Github settings for adding a new GPG key.</p>

**Comment by CryptoSalamander** [comment]
<p>I have the same problem. do you resolve it?</p>

**Comment by nuruddinshohan83** [comment]
<p>[check this out ] (<a href="https://github.com/microsoft/WSL/issues/4029" class="inline-onebox" rel="noopener nofollow ugc">WSL Ubuntu: git gpg signing Inappropriate ioctl for device · Issue #4029 · microsoft/WSL · GitHub</a>)<br>
<a href="https://github.com/microsoft/WSL/issues/4029#issuecomment-491512157" class="inline-onebox" rel="noopener nofollow ugc">WSL Ubuntu: git gpg signing Inappropriate ioctl for device · Issue #4029 · microsoft/WSL · GitHub</a></p>
